### DISCUSSION TITLE
Is it possible to resumeFromSync() with arguments?

### DISCUSSION THREAD

**Discussion by qedrohenrique** [charmplusplus/charm4py#252] [Q&A] [answered]
Let's say i have a Chare that has a method defined by:
```
def run(A):
    # ... 
```
and a resumeFromSync defined by:

```
def resumeFromSync(self):
    self.thisProxy[self.thisIndex].run()
```

How do i change resumeFromSync in order to it call run with the same A value the other Chare had?

**Comment by ZwFink** [comment, marked as answer]
There is no way to `resumeFromSync` with arguments. It's probably easiest to save `A` as a class member so `run` can access it.

**Reply by qedrohenrique** [reply]
Thanks for the answer!

Extending the question, is there a way to instantiante an `Array` with different arguments for each `Chare` in the array?

**Reply by ZwFink** [reply]
There is currently no way to do this. For the moment, the easiest way is to send all arguments to all chares in a list, and have each chare grab its respective argument.
